### PR TITLE
Add version check for antrea config fields marshal

### PR DESCRIPTION
**Pull request by hangyan**
Signed-off-by: Hang Yan <yhang@vmware.com>

### What this PR does / why we need it

This pr makes t-f will works with antrea 1.7.2/1.5.3

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

Tested on 1.24 TKR with antrea 1.5.3/antrea1.7.2 version

### Release note

```release-note

```

### CODE REVIEW

**Comment by hangyan** [comment]
@edwardbadboy

File: addons/controllers/antrea/antreaconfig_controller.go [hunk 1]
```diff
@@ -71,6 +73,12 @@ func (r *AntreaConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, nil
 	}
 
+	labels := antreaConfig.GetLabels()
+	if _, ok := labels[constants.PackageNameLabel]; !ok {
+		r.Log.Info(fmt.Sprintf("AntreaConfig resource '%v' does not contains package name label", req.NamespacedName))
+		return ctrl.Result{}, errors.New("AntreaConfig does not contains package name label")
```
Comment (edwardbadboy) [inline addons/controllers/antrea/antreaconfig_controller.go:79]: I suppose that the framework will retry automatically, right?

File: addons/controllers/antrea/antreaconfig_util.go [hunk 2]
```diff
@@ -162,6 +165,10 @@ func (r *AntreaConfigReconciler) ClusterToAntreaConfig(o client.Object) []ctrl.R
 }
 
 func mapAntreaConfigSpec(cluster *clusterv1beta1.Cluster, config *cniv1alpha1.AntreaConfig) (*antreaConfigSpec, error) {
+
+	packageName := config.GetLabels()[constants.PackageNameLabel]
+	version := strings.TrimPrefix(strings.Split(packageName, "---")[0], "antrea.tanzu.vmware.com.")
```
Comment (edwardbadboy) [inline addons/controllers/antrea/antreaconfig_util.go:170]: I think it's better to fetch version from Package spec.

File: addons/pkg/constants/constants.go [hunk 3]
```diff
@@ -235,6 +235,9 @@ const (
 	// ClusterBootstrapManagedSecret is the name for the secrets that are managed by ClusterBootstrapController
 	ClusterBootstrapManagedSecret = "clusterbootstrap-secret"
 
+	// PackageNameLabel is the label key for package name
+	PackageNameLabel = "tkg.tanzu.vmware.com/package-name"
```
Comment (edwardbadboy) [inline addons/pkg/constants/constants.go:239]: Maybe we can reuse addons/pkg/types/addons.go PackageNameLabel constant.

File: addons/controllers/antrea/antreaconfig_controller.go (hunk 1 above)
Comment (hangyan) [reply addons/controllers/antrea/antreaconfig_controller.go:81]: yes

File: addons/controllers/antrea/antreaconfig_util.go (hunk 2 above)
Comment (hangyan) [reply addons/controllers/antrea/antreaconfig_util.go:171]: this lable is also used by other component, so i think make sense to reuse it for now.

File: addons/pkg/constants/constants.go (hunk 3 above)
Comment (hangyan) [reply addons/pkg/constants/constants.go]: done

File: addons/controllers/antrea/antreaconfig_util.go [hunk 4]
```diff
@@ -208,9 +215,15 @@ func mapAntreaConfigSpec(cluster *clusterv1beta1.Cluster, config *cniv1alpha1.An
 	configSpec.Antrea.AntreaConfigDataValue.TLSCipherSuites = config.Spec.Antrea.AntreaConfigDataValue.TLSCipherSuites
 	configSpec.Antrea.AntreaConfigDataValue.DisableUDPTunnelOffload = config.Spec.Antrea.AntreaConfigDataValue.DisableUDPTunnelOffload
 	configSpec.Antrea.AntreaConfigDataValue.DefaultMTU = config.Spec.Antrea.AntreaConfigDataValue.DefaultMTU
-	configSpec.Antrea.AntreaConfigDataValue.DNSServerOverride = config.Spec.Antrea.AntreaConfigDataValue.DNSServerOverride
-	configSpec.Antrea.AntreaConfigDataValue.DisableTXChecksumOffload = config.Spec.Antrea.AntreaConfigDataValue.DisableTXChecksumOffload
-	configSpec.Antrea.AntreaConfigDataValue.EnableBridgingMode = config.Spec.Antrea.AntreaConfigDataValue.EnableBridgingMode
+
+	if semver.Compare(version, "v1.7.1") >= 0 {
+		configSpec.Antrea.AntreaConfigDataValue.MultiCluster.Enable = config.Spec.Antrea.AntreaConfigDataValue.MultiCluster.Enable
```
Comment (knabben) [inline addons/controllers/antrea/antreaconfig_util.go:220]: do we need to omit the configuration values as well / convert to pointers?

File: addons/controllers/antrea/antreaconfig_util.go (hunk 2 above)
Comment (edwardbadboy) [reply addons/controllers/antrea/antreaconfig_util.go:171]: Do you mean other component also extract the Package version according to the label value? My suggestion is that it get the Package name from the label value, and do a get call to get the Package spec, and get the version from Package spec.

Comment (hangyan) [reply addons/controllers/antrea/antreaconfig_util.go:171]: I get your point, i just mean this label is used by other part of code too, so it's kind of reliable for a while. I would like to open a catchup pr later to refactor this part, if we don't have enough time for this fix.

File: addons/controllers/antrea/antreaconfig_util.go (hunk 4 above)
Comment (hangyan) [reply addons/controllers/antrea/antreaconfig_util.go:220]: they already did, featuregate is the only ones that are missing..

**Review by knabben — APPROVED** [review]

**Review by edwardbadboy — APPROVED** [review]
The fix looks OK.

**Review by wjun — APPROVED** [review]
LGTM. Once the capz clusterclass pipeline passes management cluster creation, you can continue to add 'ok-to-merge' label and merge.

**Pull request merged**